src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.ILLink.targets

```diff
@@ -20,6 +20,11 @@ This file contains the .NET 5-specific targets to customize ILLink
           Condition=" '$(AndroidLinkMode)' == 'Full' and '%(ResolvedFileToPublish.Extension)' == '.dll' and '%(ResolvedFileToPublish.AssetType)' != 'native' "
           TrimMode="link"
       />
+      <!-- TODO: remove setting the trim mode here, once the support packages are updated to NET6 and compatability packages not needed -->
+      <ResolvedFileToPublish
+          Condition=" '$(AndroidLinkMode)' == 'SdkOnly' and ( $([System.String]::Copy(%(Filename)).StartsWith ('System.')) or $([System.String]::Copy(%(Filename)).StartsWith ('Xamarin.Android')) ) ">
```

Comment:
Did you need the `$([System.String]::Copy())` here? I think you only need it when batching also had to call something like `.Replace()`, like we did here:

https://github.com/xamarin/xamarin-android/blob/b864f93df36dc1d8dcf11b5219d37c4c7681baa3/build-tools/create-packs/Directory.Build.targets#L107

Reply:
I tried various ways to call `StartsWith` on item's `%Filename` and it didn't work for me. In the end I went with the `[System.String]` call.

Do you know a simpler expression which would work here?

Reply:
Maybe you always need it when batching... I see the `%()` now.